mysql/resource_binlog.go

```diff
@@ -0,0 +1,129 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceBinLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateBinLog,
+		UpdateContext: UpdateBinLog,
+		ReadContext:   ReadBinLog,
+		DeleteContext: DeleteBinLog,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportDatabase,
+		},
+		Schema: map[string]*schema.Schema{
+			"retention_period": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NULL",
```

Comment:
> Does the stored procedure allow fractional hours (e.g., 1.3)?

Yes, but testing with MySQL 5.7 indicates that this is converted to an int before being applied (e.g., 1.0 -> 1, 1.2 -> 1, 1.9 -> 2).

> Using a numeric type would prevent non-sensical strings from being passed into the resource.

The underlying engine (most likely) stores these as ints in any case, so we're just validating the type at the outset.

Reply:
I modified 123f9324ce48ad7947a5743d6239023837b1bbd5 to Int in this first version, so now if you declare:
```
resource "mysql_binlog_retention" "this" {
  retention_period = 1.2
}
```
It will throw an error: `Error: Attribute must be a whole number, got 1.2`